tests/conftest.py

```diff
@@ -4523,3 +4523,19 @@ def finalizer():
 
     request.addfinalizer(finalizer)
     return factory
+
+
+@pytest.fixture(scope="function", autouse=True)
+def collect_mg_rules():
+    config.REPORTING["ocs_must_gather_collection"] = True
```

Comment:
With this approach you are overwriting the configuration from the config - basically you will remove whatever will user provide in custom configuration even the default one.

Which as I pointed out is the reason why to document it if you we want to go with this approach.

@shylesh this might be a problematic in multi cluster scenario I think. Right? As we don't know which one cluster config context we are on at the moment.

Reply:
It doesn't change anything. But as this is REPORTING related variable I wold keep it under REPORTING section. Or in RUN section as it is RUN related variable which changing over the run.

Reply:
> With this approach you are overwriting the configuration from the config - basically you will remove whatever will user provide in custom configuration even the default one.
> 
> Which as I pointed out is the reason why to document it if you we want to go with this approach.
> 
> @shylesh this might be a problematic in multi cluster scenario I think. Right? As we don't know which one cluster config context we are on at the moment.

@petr-balogh , Yes , you are right. In between the multicluster tests if context gets switched then this flag will have unexpected value.